### PR TITLE
Fix #607 - Implements equality members on EventId.

### LINKED ISSUE #607: Clarify the semantics of EventId and possibly override Equals, etc.
The API of EventId makes it look like the name is optional--i.e. I can create one with just an int. But if I do this, should it be considered the same as an EventId created with the same int but also a name?

If these are supposed to represent the same event, then shouldn't Equals, etc. be implemented on the type? if they are not, then can this be documented on the class to make it clear that the name, while being optional, is significant in the semantics of the type?

### PR DESCRIPTION
**Pull request by anpete**
cc @ajcvickers

### CODE REVIEW

**Comment by anpete** [comment]
ping @BrennanConroy

**Comment by anpete** [comment]
ping @ajcvickers @Eilon

File: src/Microsoft.Extensions.Logging.Abstractions/EventId.cs [hunk 1]
```diff
@@ -5,46 +5,53 @@ namespace Microsoft.Extensions.Logging
 {
     public struct EventId
     {
-        private int _id;
-        private string _name;
-
-        public EventId(int id, string name = null)
+        public static implicit operator EventId(int i)
```
Comment (Eilon) [inline src/Microsoft.Extensions.Logging.Abstractions/EventId.cs]: `i` --> `id` ?

**Comment by muratg** [comment]
@Eilon, should we close this one?

**Comment by Eilon** [comment]
Hmm I think this is still a good change, no? @anpete ?

**Comment by anpete** [comment]
@Eilon I think so. We were waiting on an answer to https://github.com/aspnet/Logging/issues/607#issuecomment-298964897

**Comment by Eilon** [comment]
I pingo-pongo'ed that thread, thanks!

**Comment by ajcvickers** [comment]
@Eilon Note that this is a breaking change for any code that is relying on the existing behavior.

**Comment by Eilon** [comment]
Ah, fair point. But if we had to guess, it probably wouldn't *really* break anyone, right? I'd be surprised if many people are even testing for equality on these instances, and even if they are, they probably don't realize their code doesn't work... would you agree?

**Comment by ajcvickers** [comment]
@Eilon Yes, I agree.

**Comment by davidfowl** [comment]
I think the bar should be super high for braking changes with the Microsoft.Extensions.* as they have been widely adopted by the broader .net ecosystem. That said, this looks benign.

**Comment by Eilon** [comment]
@ajcvickers - ok.
@davidfowl - I agree about the bar, and I think in this case it's acceptable.

**Comment by anpete** [comment]
OK, will get this in.

**Comment by davidfowl** [comment]
@anpete this done?

**Comment by anpete** [comment]
@davidfowl Fell off my radar. We still want it?

**Comment by davidfowl** [comment]
Yea, it looks harmless. Do you guys need it still?

**Comment by davidfowl** [comment]
@Eilon @anpete lets close or merge.

**Pull request merged**